wgpu-hal/src/gles/command.rs

```diff
@@ -0,0 +1,871 @@
+    }
+    unsafe fn set_blend_constants(&mut self, color: &wgt::Color) {
+        let color = [
+            color.r as f32,
+            color.g as f32,
+            color.b as f32,
+            color.a as f32,
+        ];
+        self.cmd_buffer.commands.push(C::SetBlendConstant(color));
+    }
+
+    unsafe fn draw(
+        &mut self,
+        start_vertex: u32,
+        vertex_count: u32,
+        start_instance: u32,
+        instance_count: u32,
+    ) {
+        self.prepare_draw(start_instance);
+        self.cmd_buffer.commands.push(C::Draw {
+            topology: self.state.topology,
+            start_vertex,
+            vertex_count,
+            instance_count,
+        });
+    }
+    unsafe fn draw_indexed(
+        &mut self,
+        start_index: u32,
+        index_count: u32,
+        base_vertex: i32,
+        start_instance: u32,
+        instance_count: u32,
+    ) {
+        self.prepare_draw(start_instance);
+        let (index_size, index_type) = match self.state.index_format {
+            wgt::IndexFormat::Uint16 => (2, glow::UNSIGNED_SHORT),
+            wgt::IndexFormat::Uint32 => (4, glow::UNSIGNED_INT),
+        };
+        let index_offset = self.state.index_offset + index_size * start_index as wgt::BufferAddress;
+        self.cmd_buffer.commands.push(C::DrawIndexed {
+            topology: self.state.topology,
+            index_type,
+            index_offset,
+            index_count,
+            base_vertex,
+            instance_count,
+        });
+    }
+    unsafe fn draw_indirect(
+        &mut self,
+        buffer: &super::Buffer,
+        offset: wgt::BufferAddress,
+        draw_count: u32,
+    ) {
+        self.prepare_draw(0);
+        for draw in 0..draw_count as wgt::BufferAddress {
+            let indirect_offset =
+                offset + draw * mem::size_of::<wgt::DrawIndirectArgs>() as wgt::BufferAddress;
+            self.cmd_buffer.commands.push(C::DrawIndirect {
+                topology: self.state.topology,
+                indirect_buf: buffer.raw,
+                indirect_offset,
+            });
+        }
+    }
+    unsafe fn draw_indexed_indirect(
+        &mut self,
+        buffer: &super::Buffer,
+        offset: wgt::BufferAddress,
+        draw_count: u32,
+    ) {
+        self.prepare_draw(0);
+        let index_type = match self.state.index_format {
+            wgt::IndexFormat::Uint16 => glow::UNSIGNED_SHORT,
+            wgt::IndexFormat::Uint32 => glow::UNSIGNED_INT,
+        };
+        for draw in 0..draw_count as wgt::BufferAddress {
+            let indirect_offset = offset
+                + draw * mem::size_of::<wgt::DrawIndexedIndirectArgs>() as wgt::BufferAddress;
+            self.cmd_buffer.commands.push(C::DrawIndexedIndirect {
+                topology: self.state.topology,
+                index_type,
+                indirect_buf: buffer.raw,
+                indirect_offset,
+            });
+        }
+    }
+    unsafe fn draw_indirect_count(
```

Comment:
Unless we can use the desktop gl ext/gl4.6 function here, this should probably just be unreachable! as we won't ever be able to implement them.